### ISSUE TITLE
[FlyteCTL Feature] Archive a specific workflow version

### ISSUE THREAD

**Issue by LukasBommes** [flyteorg/flyte#3869]
### Describe the feature/command for FlyteCTL

Currently, `flytectl update workflow-meta` can only be used to archive an entire workflow (all versions). There is no option to archive a specific version of a workflow.

See this slack thread: https://flyte-org.slack.com/archives/CP2HDHKE1/p1689084246507979

### Provide a possible output or UX example

We need support for the `--version` flag in `flytectl update workflow-meta`:

```
flytectl update workflow-meta -p flytesnacks -d development --version <myversion> core.control_flow.merge_sort.merge_sort --archive
```

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by pingsutw** [comment]
cc @Hacker-Davinci would you like to work on it?

**Comment by Future-Outlier** [comment]
Yes I am interested at it, please give me a chance, thanks a lot kevin and flyte community

**Comment by eapolinario** [comment]
@Future-Outlier , I just assigned this to you. Please let me know if you need help.

**Comment by Future-Outlier** [comment]
Sure, thanks a lot!

**Comment by kumare3** [comment]
@Future-Outlier i don't think this can be done, shall we close this?

**Comment by Future-Outlier** [comment]
> @Future-Outlier i don't think this can be done, shall we close this?

Why? I haven't deep dived into it, but can try it after 5/12(have a talk to give).

**Comment by Future-Outlier** [comment]
> > @Future-Outlier i don't think this can be done, shall we close this?
> 
> Why? I haven't deep dived into it, but can try it after 5/12(have a talk to give).

I'll leave my sight after deep diving it, thank you.
